src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,101 @@
+import { useParams } from 'react-router-dom';
+import { NavLink } from 'react-router-dom';
+import { Person } from '../../types/Person';
+import cn from 'classnames';
+
+enum Properties {
+  Name = 'Name',
+  Sex = 'Sex',
+  Born = 'Born',
+  Died = 'Died',
+  Mother = 'Mother',
+  Father = 'Father',
+}
+
+interface Props {
+  tablePeople: Person[];
+}
+
+export const PeopleTable: React.FC<Props> = props => {
+  const { tablePeople } = props;
+
+  const { slug: prevSlug } = useParams();
+
+  const findMotherAndFather = (nameOfParent: string | null): Person | null => {
+    return tablePeople.find(person => person.name === nameOfParent) || null;
+  };
+
+  const selectedPeople = tablePeople.map(person => {
+    return {
+      ...person,
+      father: findMotherAndFather(person.fatherName),
+      mother: findMotherAndFather(person.motherName),
+    };
+  });
```

Comment:
Do it in the parent(container) component. `PeopleTable` is a presentational component in your case so avoid keeping any logic in it
https://www.freecodecamp.org/news/separation-of-concerns-react-container-and-presentational-components/